File: t/VB/OmrProcessingDemo/UI/CloseRequestArgs.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Public Class CloseRequestArgs
   Inherits EventArgs

   Public Property ClosingState As ClosingReason

   Public Enum ClosingReason
      RevertToIntro
      ExplicitNew
      ToLoadExisting
      ApplicationExiting
   End Enum

   Public Sub New(ByVal closeState As ClosingReason)
      MyBase.New()
      Me.ClosingState = closeState
   End Sub
End Class

File: t/CS/OmrProcessingDemo/UI/CloseRequestArgs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OmrProcessingDemo.UI
{
   public class CloseRequestArgs : EventArgs
   {
      public ClosingReason ClosingState { get; private set; }

      public enum ClosingReason
      {
         RevertToIntro,
         ExplicitNew,
         ToLoadExisting,
         ApplicationExiting
      }

      public CloseRequestArgs(ClosingReason closeState) : base()
      {
         this.ClosingState = closeState;
      }
   }
}
